builtin/logical/database/backend.go

```diff
@@ -27,6 +30,32 @@ const (
 	minRootCredRollbackAge = 1 * time.Minute
 )
 
+// mutable for tests only
+var cleanupMaxWaitTime = 500 * time.Millisecond
+
+// metrics collection
+var (
+	gaugeSync = sync.Mutex{}
+	gauges    = map[string]*int32{}
```

Comment:
I'm always discomforted when I see non-readonly globals, can we pull this into databaseBackend instead?  I'm also not sure we need a generic gauge manager, why not just have an atomic `numConnections` field in databaseBackend?  That way we won't have to worry about the lock to protect the map and any possible unpleasantness it might cause us.

Reply:
A single `numConnections` wouldn't allow us to see the different types of databases though.

I wanted to avoid putting these into `databaseBackend` because we might have multiple `databaseBackend`s running, and we probably would like to aggregate these gauges across all of them. With gauges (instead of counters), I don't *think* that `armon/go-metrics` or `statsd` will automatically aggregate them for us (whereas with counters, I believe `statsd` at least would).

Reply:
Ideally, I would like to add a tag with the instance of the database backend -- maybe the mount path? But I don't see any way to get that. We could tag by the BackendUUID though.